kolibri/plugins/user_auth/assets/src/views/DeviceUnusableMessage.vue

```diff
@@ -0,0 +1,135 @@
+
+  import { computed, ref, watch } from 'vue';
+  import { DeviceUnusableReason } from '../constants';
+
+  const DEPROVISION_TASK_QUEUE = 'device_deprovision';
+
+  const strings = createTranslator('DeviceUnusableStrings', {
+    noSuperusersNotice: {
+      message: 'This device is unusable because there are no superuser accounts on this device.',
+      context: 'Notice that there are no superuser accounts on the device',
+    },
+    noSuperuserCallToAction: {
+      message: 'Please reinstall Kolibri to create a superuser account.',
+      context: 'Call to action to reinstall Kolibri to create a superuser account',
+    },
+    superusersSoftDeletedNotice: {
+      message:
+        'This device is unusable because all superuser accounts on this device have been deleted on the server.',
+      context: 'Notice that all superuser accounts have been deleted on the server',
+    },
+    superusersSoftDeletedCallToAction: {
+      message:
+        'Please contact your system administrator to restore your account or reinstall Kolibri to create a new superuser account.',
+      context:
+        'Call to action to contact system administrator or reinstall Kolibri to create a new superuser account',
+    },
+    unknownIssueNotice: {
+      message: 'This device is unusable due to an unknown reason. Please contact support.',
+      context: 'Notice that the device is unusable due to an unknown reason',
+    },
+    reinstallKolibriAction: {
+      message: 'Reinstall Kolibri',
+      context: 'Button text to reinstall Kolibri',
+    },
+    deprovisioningError: {
+      message: 'An error occurred while trying to reinstall Kolibri. Please try again.',
+      context: 'Error message when there is an error deprovisioning the device',
+    },
+  });
+
+  export default {
+    name: 'DeviceUnusableMessage',
+    setup() {
+      const taskLoading = ref(false);
+      const { tasks } = useTaskPolling(DEPROVISION_TASK_QUEUE);
+      const { createSnackbar } = useSnackbar();
+      const deprovisionTask = computed(() => tasks.value[tasks.value.length - 1]);
+
+      const deprovision = async () => {
+        try {
+          taskLoading.value = true;
+          await TaskResource.startTask({
+            type: TaskTypes.DEPROVISIONDEVICE,
+          });
+        } catch (e) {
+          createSnackbar(strings.deprovisioningError$());
+        }
+      };
+
+      const clearTasks = async () => {
+        try {
+          await TaskResource.clearAll(DEPROVISION_TASK_QUEUE);
+        } catch (e) {
+          return;
+        }
+      };
+
+      watch(deprovisionTask, task => {
+        if (!task) return;
+        taskLoading.value = true;
+        if (task.status === TaskStatuses.FAILED) {
+          taskLoading.value = false;
+          createSnackbar(strings.deprovisioningError$());
+          clearTasks();
+          return;
+        }
+        if (task.status === TaskStatuses.COMPLETED) {
+          clearTasks();
+          redirectBrowser();
```

Comment:
When the browser redirects here is there a chance we'd end up cancelling/clobbering the async function call through page navigation? This initially stuck out to me because you call an async function synchronously which seems less problematic above, but here I wonder if we might miss out on clearing the tasks if these run synchronously